power: add DEVICE_PM_LOW_POWER_STATE for device power management

wentongwu opened the pull request:
when system going to sleep state(not deep sleep), make peripherals
go to DEVICE_PM_LOW_POWER_STATE state which need less time than
DEVICE_PM_SUSPEND_STATE to save more power in the sleep state.

Fixes: #17838.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

pdunaj commented on subsys/power/power.c:
```diff
@@ -116,6 +116,16 @@ enum power_states _sys_suspend(s32_t ticks)
+		if (sys_pm_low_power_devices()) {
+			LOG_ERR("Device enter low power state failed!");
+			sys_pm_notify_power_state_exit(pm_state);
+			pm_state = SYS_POWER_STATE_ACTIVE;
```
This is not really an active state as it could happen after some devices were set to low power mode.
Unless we try to revert (activate) all powered down devices when error occur while traversing list in `sys_pm_low_power_devices`

wentongwu replied:
@pdunaj Thanks for the review, yes, that should be a problem. Please review the new patch. Thanks

pdunaj commented on subsys/power/device.c:
```diff
@@ -66,6 +66,24 @@ int sys_pm_suspend_devices(void)
+		if (device_retval[i]) {
+			LOG_ERR("%s low power operation failed\n",
+					pm_device_list[idx].config->name);
+			return device_retval[i];
```
see comment later - maybe we should reactivate devices powered down before the error occurred?

wentongwu committed:
power: add DEVICE_PM_LOW_POWER_STATE for device power management

when system going to sleep state(not deep sleep), make peripherals
go to DEVICE_PM_LOW_POWER_STATE state which need less time than
DEVICE_PM_SUSPEND_STATE to save more power in the sleep state.

Fixes: #17838.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

wentongwu committed:
power: active all powered down devices when suspend devices failed

when suspend devices failed, system will still keep in active state,
so resume all the suspended devices.

Fixes: #17838.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>
